MyPlayground.playground/Contents.swift

```diff
@@ -0,0 +1,80 @@
+func mean(_ numbers: [Int]) -> Double? {
+    if numbers.isEmpty { return nil }
+    
+    var sumOfNumbers = 0
+    
+    for number in numbers {
+        sumOfNumbers += number
+    }
+
+    let mean = Double(sumOfNumbers) / Double(numbers.count)
+
+    return mean
+}
+
+
+func median(_ numbers: [Int]) -> Double? {
+    if numbers.isEmpty { return nil }
+    
+    let sortedNumbers = numbers.sorted()
+    let middleIndex = sortedNumbers.count / 2
+    
+    if sortedNumbers.count.isMultiple(of: 2) {
+        let median = Double(sortedNumbers[middleIndex] + sortedNumbers[middleIndex - 1]) / 2
+        return median
+    } else {
+        let median = Double(sortedNumbers[middleIndex])
+        return median
+    }
+}
+
+
+func modes(_ numbers: [Int]) -> [Int] {
+    
+    if numbers.isEmpty { return [] }
+
+    var counter = [Int: Int]()
+    var modes = Set<Int>()
+
+    for number in numbers {
+        counter[number] = counter[number, default: 0] + 1
+    }
+
+    let sortedCounters = counter.sorted { return $0.value > $1.value }
+    var maxValue = 1
+
+    for (key, value) in sortedCounters {
+        if maxValue > value {
+            break
+        } else {
+            modes.insert(key)
+            maxValue = value
+        }
+    }
+    
+    return modes.sorted()
+}
+
+
+let testNumbers = [1,2,3,8,2,3,2,4,4,4]
+
+
+if let mean = mean(testNumbers) {
+    print("Mean: \(mean)")
+} else {
+    print("There is no mean value in empty array.")
+}
+
+if let median = median(testNumbers) {
+    print("Median: \(median)")
+} else {
+    print("There is no median value in empty array.")
+}
+
+let valueModes = modes(testNumbers)
+
+if !valueModes.isEmpty {
+    print("Modes: \(valueModes)")
+} else {
+    print("There is no modes value in empty array.")
```

Comment:
Typo (due to plural):

```suggestion
    print("There is no mode values in empty array.")
```

or

```suggestion
    print("There is no modes in empty array.")
```

Reply:
Fixed.